app/src/main/java/com/concordium/wallet/core/arch/BackendEventObserver.kt

```diff
@@ -12,24 +12,19 @@ import com.concordium.wallet.core.backend.BackendEventResource
 
 open class BackendEventObserver<T> : Observer<BackendEventResource<T>> {
 
-    override fun onChanged(backendResource: BackendEventResource<T>?) {
+    override fun onChanged(backendResource: BackendEventResource<T>) {
```

Comment:
I can tell that this change is necessary for the code to compile now, but I don't see why. Can you help me understand? What changed for this to be the case?

Reply:
the value in super fun is not nullable so it should not be here. 
![image](https://github.com/Concordium/concordium-reference-wallet-android/assets/11137601/e1aa9c58-2fcd-4084-8ed8-015ded62c9f3)

Reply:
But then why did it work before? From what I can tell this interface didn't change?